doc/specs/#1012 - Show dependencies.md

```diff
@@ -0,0 +1,111 @@
+* Windows Libraries
+* Package Dependencies (same source)
+* External Dependencies
+
+Only dependencies declared in the manifest will be shown, not the entire dependency graph.
+
+### install
+The install command will enumerate the dependencies of the package version being installed as follows:
+```
+> winget install Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+This package requires the following dependencies:
+  - Windows Feature: Hyper-V
+  - Package: Microsot.WindowsTerminal
+Downloading https://github.com/notepad-plus-plus/notepad-plus-plus/releases/download/v7.9.5/npp.7.9.5.Installer.x64.exe
+Successfully verified installer hash
+Starting package install...
+```
+
+### show
+The show command will enumerate the dependencies of the package as follows:
+```
+> winget show Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+Version: 7.9.5
+Publisher: Notepad++ Team
+Author: Don Ho
+Moniker: notepad++
+Description: Notepad++ is a free (as in “free speech” and also as in “free beer”) source code editor and Notepad replacement that supports several languages. Running in the MS Windows environment, its use is governed by GNU General Public License.
+Homepage: https://notepad-plus-plus.org/
+License: GPL-2.0-only
+License Url: https://raw.githubusercontent.com/notepad-plus-plus/notepad-plus-plus/v7.9.5/LICENSE
+Installer:
+  Type: Nullsoft
+  Locale: en-US
+  Download Url: https://github.com/notepad-plus-plus/notepad-plus-plus/releases/download/v7.9.5/npp.7.9.5.Installer.x64.exe
+  SHA256: 4881548cd86491b453520e83c19292c93b9c6ce485a1f9eb9301e3913a9baced
+  Dependencies:
+    - Windows Feature: Hyper-V
+    - Package: Microsot.WindowsTerminal
+```
+
+### upgrade
+The upgrade command will enumerate the dependencies of the package as follows:
+```
+> winget upgrade Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+This package requires the following dependencies:
+  - Windows Feature: Hyper-V
+  - Package: Microsot.WindowsTerminal
+Successfully verified installer hash
+Starting package install...
+```
+As of now, it will not try to validate nor install any of the dependencies for any package version.
+
+### uninstall
+```
+> winget uninstall Notepad++
+Found Notepad++ [Notepad++.Notepad++]
+Package had dependencies that may not be needed anymore:
+  - Windows Feature: Hyper-V
+  - Package: Microsot.WindowsTerminal
+Starting package uninstall...
+Successfully uninstalled
+```
+Dependencies will not be uninstalled.
+
+### validate
+For this step dependencies will not be validated. 
+
+### import
+Will show dependencies before installing each of the specified packages on the import; i.e., it will not gather all the dependencies and show them at once before starting.

```

Comment:
Why not gather all dependencies?  I agree that it isn't strictly needed, but it seems to go against the goal of having the majority of the code path laid out by this change.

Reply:
Well it's something I wanted to ask. Import should install packages only if it can install them all? If so then yes, gathering all the dependencies is the best approach. Otherwise, it may be better to check and install one by one. 

Although now that I think about it import has an option for what to do right?(--ignore-unavailable) maybe it can be different depending on that?

Reply:
If I remember the default behavior correctly, `import` will fail before doing any work if it cannot provide one of the packages.  To maintain this same behavior, all dependencies would also need to be verified before attempting any installations.